### PR TITLE
Fix Agit pull request permission check

### PR DESCRIPTION
**Pull request by a1012112796**
user with read permission should also can create agit flow pull request. looks this logic was broken in https://github.com/go-gitea/gitea/pull/31033 this pull request try fix it and add test code.

### CODE REVIEW

File: services/pull/pull.go [hunk 1]
```diff
@@ -64,7 +64,7 @@ func NewPullRequest(ctx context.Context, opts *NewPullRequestOptions) error {
 	}
 
 	// user should be a collaborator or a member of the organization for base repo
-	if !issue.Poster.IsAdmin {
+	if (!issue.Poster.IsAdmin) && pr.Flow == issues_model.PullRequestFlowGithub {
```
Comment (lunny) [inline services/pull/pull.go]: Maybe you should only change for line 82?

Comment (a1012112796) [reply services/pull/pull.go]: > Maybe you should only change for line 82?

why？ looks the result is same.

Comment (wxiaoguang) [inline services/pull/pull.go]:
```suggestion
	canCreate := issue.Poster.IsAdmin || pr.Flow == issues_model.PullRequestFlowAGit
	if !canCreate {
```

But should we also check something like "repo read permission" for "agit" to create?

Comment (a1012112796) [reply services/pull/pull.go]: > But should we also check something like "repo read permission" for "agit" to create?

looks user already can't create pull request in my local test.

image

**Review by wxiaoguang — APPROVED** [review]

**Review by hiifong — APPROVED** [review]

**Review by lunny — APPROVED** [review]

**Pull request merged**